clean up room viewer urls

majestrate opened the pull request:
seperate room viewer routes into its own file

Commit:
clean up room viewer routes

* refactor urls to be nicer
* clean up room model code to include url property for viewer
* dry out the code

jagerman commented on sogs/utils.py:
```diff
@@ -102,7 +102,7 @@ def bencode_consume_string(body: memoryview) -> Tuple[memoryview, memoryview]:
 
 def server_url(room):
-    return '{}/{}?public_key={}'.format(config.URL_BASE, room or '', crypto.server_pubkey_hex)
+    return '{}/r/{}?public_key={}'.format(config.URL_BASE, room or '', crypto.server_pubkey_hex)
```
This is going to have to be delayed until it gets supported by Session, unfortunately.

jagerman replied:
Either revert this (and we'll apply it later, once Session is ready), or perhaps make a `config.` global for whether or not we add the `/r` here.

jagerman reviewed (approved):
One tweak to not deploy the `/r` yet in the open group URL, other than that looks good.

jagerman committed:
Undo /r/ in server_url for now (leave a TODO comment)

We can't enable this until Session clients support it (because that is
the main consumer of *this* particular URL value).
